dftimewolf/metawolf/utils.py

```diff
@@ -0,0 +1,361 @@
+        if recipe == LAST_ACTIVE_RECIPE:
+          # In this case, settables holds the value of the latest used recipe
+          # and is not a dictionary
+          loaded_sessions[session_id][LAST_ACTIVE_RECIPE] = settables
+          continue
+        if recipe == LAST_ACTIVE_PROCESSES:
+          # In this case, settables holds the list of processes run in the
+          # session
+          loaded_sessions[session_id][LAST_ACTIVE_PROCESSES] = settables
+          continue
+        loaded_sessions[session_id][recipe] = {}
+        for settable_id, settable in settables.items():
+          loaded_sessions[session_id][recipe][settable_id] = self.Unmarshal(
+              settable) if unmarshal else settable
+
+    return loaded_sessions
+
+  def RunInBackground(
+      self,
+      processes: List['output.MetawolfProcess']
+  ) -> bool:
+    """Guard executed before `quit` or SIGINT.
+
+    If processes are still running, ask the user if they want to let them run
+    in the background.
+
+    Args:
+      processes (List[output.MetawolfProcess]): The processes associated to
+          Metawolf's session.
+
+    Returns:
+      bool: True if the processes should keep running in the background.
+    """
+    still_running = False
+    for metawolf_process in processes:
+      if metawolf_process.Poll() is None:
+        still_running = True
+        break
+    if still_running:
+      value = input('Metawolf still has running processes. Would you like to '
+                    'keep them running in the background [Yn]? ') or 'y'
+      q = self.Str2Bool(str(value))
+      while q not in [False, True]:
+        value = input('[Yn]? ') or 'y'
+        q = self.Str2Bool(str(value))
+      return q
+    return True
+
+  def PrepareDFTimewolfCommand(
+      self,
+      recipe: str,
+      session_settables: Dict[str, session.SessionSettable]
+  ) -> List[str]:
+    """Builds a cli command from current session parameters.
+
+    Args:
+      recipe (str): The name of the recipe to run.
+      session_settables (Dict[str, SessionSettable]): The dictionary holding the
+          session settables to use to build the command.
+
+    Returns:
+      List[str]: A list that contains the CLI command elements based on the
+          current recipe and settables.
+    """
+    cmd_components = {}
+    for _, settable in session_settables.items():
+      if settable.name == SESSION_ID_SETTABLE:
+        continue
+      if not settable.IsOptional():
+        value = settable.GetValue()
+        if value is None:
+          return []
+        cmd_components[settable.name] = '{0!s}'.format(value)
+      else:
+        value = settable.GetValue()
+        if value:
+          cmd_components[settable.name] = '--{0:s}={1!s}'.format(
+              settable.name, value)
+
+    # We need to order the arguments as they are defined in the recipe
+    cmd = [DFTIMEWOLF, recipe]
+    for name, _, _ in self.recipe_manager.Recipes()[recipe].args:
```

Comment:
I don't see any `Recipe()` function in RecipeManager, only `GetRecipes()` which I guess is what you're trying to call here (and what you do call in L 342)

Reply:
There is! Checkout `dftimewolf/lib/recipes/manager.py` around l127 in the changes.